src/unspent_store.py

```diff
@@ -0,0 +1,204 @@
+                            block.height, 1, removed.coinbase) # type: ignore # noqa
+            diff_store.diffs[spent.name.hex()] = spent
+
+        for coin in additions:
+            added: Unspent = Unspent(coin, block.height, 0, 0, 0) # type: ignore # noqa
+            diff_store.diffs[added.name.hex()] = added
+
+        coinbase: Unspent = Unspent(block.body.coinbase, block.height, 0, 0, 1) # type: ignore # noqa
+        diff_store.diffs[coinbase.name.hex()] = coinbase
+        fees_coin: Unspent = Unspent(block.body.fees_coin, block.height, 0, 0, 1) # type: ignore # noqa
+        diff_store.diffs[fees_coin.name.hex()] = fees_coin
+
+    # Store unspent in DB and ram cache
+    async def add_unspent(self, unspent: Unspent) -> None:
+        cursor = await self.unspent_db.execute(
+            "INSERT OR REPLACE INTO unspent VALUES(?, ?, ?, ?, ?, ?)",
+            (unspent.coin.name().hex(),
+             unspent.confirmed_block_index,
+             unspent.spent_block_index,
+             int(unspent.spent),
+             int(unspent.coinbase),
+             bytes(unspent)),
+        )
+        await cursor.close()
+        await self.unspent_db.commit()
+        self.lca_unspent_coins[unspent.coin.name().hex()] = unspent
+        if len(self.lca_unspent_coins) > 600000:
+            while len(self.lca_unspent_coins) > 600000:
+                first_in = list(self.lca_unspent_coins.keys())[0]
+                del self.lca_unspent_coins[first_in]
+
+    # Update unspent to be spent in DB
+    async def set_spent(self, coin_name: bytes32, index: uint32):
+        current: Optional[Unspent] = await self.get_unspent(coin_name)
+        if current is None:
+            return
+        spent: Unspent = Unspent(current.coin, current.confirmed_block_index,
+                                 index, 1, current.coinbase) # type: ignore # noqa
+        await self.add_unspent(spent)
+
+    # Checks DB and DiffStores for unspent with coin_name and returns it
+    async def get_unspent(self, coin_name: CoinName, header: HeaderBlock = None) -> Optional[Unspent]:
+        if header is not None and header.header_hash in self.head_diffs:
+            diff_store = self.head_diffs[header.header_hash]
+            if coin_name.hex() in diff_store.diffs:
+                return diff_store.diffs[coin_name.hex()]
+        if coin_name.hex() in self.lca_unspent_coins:
+            return self.lca_unspent_coins[coin_name.hex()]
+        cursor = await self.unspent_db.execute(
+            "SELECT * from unspent WHERE coin_name=?", (coin_name.hex(),)
+        )
+        row = await cursor.fetchone()
+        await cursor.close()
+        if row is not None:
+            return Unspent.from_bytes(row[5])
+        return None
+
+    # TODO figure out if we want to really delete when doing rollback
+    async def rollback_lca_to_block(self, block_index):
+        # Update memory cache
+        for k in list(self.lca_unspent_coins.keys()):
```

Comment:
you can do:
for coin_name, unspent in self.lca_unspent_coins.items():
    .....
It's better style